### ISSUE TITLE
Doesn't recognize web component buttons as buttons

### ISSUE THREAD

**Issue by jchen790** [microsoft/accessibility-insights-web#4385]
## Describe the bug

When running the Accessibility Insights tool on a page that uses button web components, the tool doesn't recognize them as buttons.

**To Reproduce**

1. Open a web page using button web components. The button web components must have a closed shadow DOM and should have aria-label set.
2. Run the Accessibility Insights tool
3. See that the tool will state the button web components shouldn't use ARIA attributes not allowed for its role, specifically calling out aria-label.

## Expected behavior

Tool should recognize these button web components as buttons and not mark with issue that aria-label shouldn't be used for these components.

## Context (please complete the following information)

- OS Name & Version: Windows 10
- AI-Web Version & Environment: 2.27.0 production
- Browser Version: Edge Version 91.0.864.41

**Comment by lamaks** [comment]
@jchen790 Please provide link to a web page with bug repro
